### PR TITLE
libsql: Add initial support

### PR DESCRIPTION
**Pull request by Ty3uK**
libsql code heavely based on sqlite code.

Database drop was written manually because you cannot remove remote database.

### CODE REVIEW

**Comment by amacneil** [comment]
Never heard of libQSL before. The [github page](https://github.com/libsql/libsql) says that they will maintain 100% compatibility with the file format and API, so why is a new driver needed?

**Comment by Ty3uK** [comment]
> Never heard of libQSL before. The [github page](https://github.com/libsql/libsql) says that they will maintain 100% compatibility with the file format and API, so why is a new driver needed?

Because libSQL also allows you to use remote db over http/ws/grpc.

Check [Turso](https://turso.tech/)

File: pkg/driver/libsql/libsql.go [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+	"github.com/lib/pq"
+	_ "github.com/libsql/libsql-client-go/libsql" // database/sql driver
+)
+
+func init() {
+	dbmate.RegisterDriver(NewDriver, "libsql")
+	dbmate.RegisterDriver(NewDriver, "http")
+	dbmate.RegisterDriver(NewDriver, "https")
+	dbmate.RegisterDriver(NewDriver, "ws")
+	dbmate.RegisterDriver(NewDriver, "wss")
```
Comment (amacneil) [inline pkg/driver/libsql/libsql.go:26]: huh? why is it accessible over http + websocket?

Comment (Ty3uK) [reply pkg/driver/libsql/libsql.go:26]: Check this out 🙂

https://github.com/libsql/sqld

**Comment by emilpriver** [comment]
Hey, Just wondering if there is any update on this topic? :D

**Comment by crvouga** [comment]
Is there any updates on this? Will it be added soon?

File: pkg/driver/libsql/libsql.go (hunk 1 above)
Comment (dossy) [reply pkg/driver/libsql/libsql.go:26]: Registering the very generic `http`, `https`, `ws` and `wss` to this driver is probably going to bite us in the future.  I would rather see us register `libsql+http`, `libsql+https`, `libsql+ws` and `libsql+wss` and peel off the `libsql+` prefix inside `ConnectionString()`.

File: pkg/driver/libsql/libsql.go [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+	}
+
+	migrations, err := drv.schemaMigrationsDump(db)
+	if err != nil {
+		return nil, err
+	}
+
+	schema = append(schema, migrations...)
+	return dbutil.TrimLeadingSQLComments(schema)
+}
```
Comment (dossy) [inline pkg/driver/libsql/libsql.go:141]: This doesn't currently work:

```
# dist/dbmate -u $LIBSQL_TEST_URL -d testdata/db/migrations dump
Error: Error: 'tls' usage forbidden. Please use 'WithTls' option instead
```

**Pull request closed without merging**